pyquil/api/_compiler.py

```diff
@@ -129,71 +129,87 @@ def _collect_memory_descriptors(program: Program) -> Dict[str, ParameterSpec]:
 class QPUCompiler(AbstractCompiler):
     @_record_call
     def __init__(self,
-                 endpoint: str,
+                 quilc_endpoint: str,
+                 qpu_compiler_endpoint: Optional[str],
                  device: AbstractDevice,
                  timeout: int = 10,
                  name: Optional[str] = None) -> None:
         """
         Client to communicate with the Compiler Server.
 
-        :param endpoint: TCP or IPC endpoint of the Compiler Server
+        :param quilc_endpoint: TCP or IPC endpoint of the Quil Compiler (quilc)
+        :param qpu_compiler_endpoint: TCP or IPC endpoint of the QPU Compiler
         :param device: PyQuil Device object to use as compilation target
         :param timeout: Number of seconds to wait for a response from the client.
         :param name: Name of the lattice being targeted
         """
 
-        if not endpoint.startswith('tcp://'):
+        if not quilc_endpoint.startswith('tcp://'):
             raise ValueError(f"PyQuil versions >= 2.4 can only talk to quilc "
                              f"versions >= 1.4 over network RPCQ.  You've supplied the "
-                             f"endpoint '{endpoint}', but this doesn't look like a network "
+                             f"endpoint '{quilc_endpoint}', but this doesn't look like a network "
                              f"ZeroMQ address, which has the form 'tcp://domain:port'. "
                              f"You might try clearing (or correcting) your COMPILER_URL "
                              f"environment variable and removing (or correcting) the "
                              f"compiler_server_address line from your .forest_config file.")
 
-        self.client = Client(endpoint, timeout=timeout)
+        self.quilc_client = Client(quilc_endpoint, timeout=timeout)
+        if qpu_compiler_endpoint is not None:
+            self.qpu_compiler_client = Client(qpu_compiler_endpoint, timeout=timeout)
+        else:
+            self.qpu_compiler_client = None
```

Comment:
Do we want to warn here, or anything?

Reply:
Probably a good idea; I can't see this getting hit in any good code path.

Reply:
Well, I think this would be hit if a user instantiates a quantum computer with `get_qc('lattice_name')` while they are not engaged. It should be okay to do that (I think), except they should be warned that they ought to add a `-qvm` if they want to do anything with it right away

Reply:
This should be fixed at the point of calling get_qc where there is more context on what the user was trying to do. The below warning requires a good amount of knowledge of how PyQuil + QCS config works under the hood